src/service.rs

```diff
@@ -487,17 +527,24 @@ impl Account {
 
     fn main_crypto_manager(&self) -> Result<MainCryptoManager> {
         let version = self.version;
-        let main_key = &self.main_key;
-        MainCryptoManager::new(try_into!(&main_key[..])?, version)
+        MainCryptoManager::new(&self.main_key, version)
     }
 
     fn identity_crypto_manager(&self) -> Result<BoxCryptoManager> {
         let main_crypto_manager = self.main_crypto_manager()?;
         let content = main_crypto_manager
             .0
             .decrypt(&self.user.encrypted_content, None)?;
-        let privkey = &content[SYMMETRIC_KEY_SIZE..];
-        main_crypto_manager.identity_crypto_manager(try_into!(privkey)?)
+
+        // The content is the concatenation of the account key and the private key
+        let privkey = content
+            .get(SYMMETRIC_KEY_SIZE..(SYMMETRIC_KEY_SIZE + PRIVATE_KEY_SIZE))
```

Comment:
Why the math changed? Does get not support open ended ranges?

Reply:
The indexing needs to be exact so that the `.try_into()` below can't fail. If `.get(SYMMETRIC_KEY_SIZE..)` was used here, it would only catch a slice that's shorter than `SYMMETRIC_KEY_SIZE`; if it had any other incorrect size, trying to convert the result to an array of length `PRIVATE_KEY_SIZE` below would still fail, and would need separate error handling.

https://github.com/rust-lang/rust/issues/90091 should make all these things a lot prettier if it ever lands.

Reply:
Cool.